GitTools/GitVersion
# Add BuildKite build agent

## Description
Added a Buildkite build agent so that GitVersion can use the `BUILDKITE_BRANCH` environment variable when running on a Buildkite agent.

## Related Issue

Resolves #2956.

## Motivation and Context
By default Buildkite checkout a branch in a headless state. This change removes the need for adding this at the start of a Buildkite pipeline:
```shell
git checkout $BUILDKITE_BRANCH
git reset HEAD --hard $BUILDKITE_COMMIT
```
## How Has This Been Tested?
- Added unit tests
- Manually tested in a Buildkite pipeline

Please note I wasn't able to check my added documentation is valid as I got this error trying to run the build script:
```
❯ .\build.ps1 -Target Preview-Documentation
Running build stage build...

Tool Codecov.Tool is already installed, with required version.
Tool GitVersion.Tool is already installed, with required version.
Error: The target 'Preview-Documentation' was not found.
```
## Checklist:
